### DISCUSSION TITLE
Non-geoblocked over geo-blocked policy

### DISCUSSION THREAD

**Discussion by Dum4G** [iptv-org/community#503] [Ideas]
I just noticed that some of geo-blocked sources are currently not used for channels that have any other source. That leads to using questionable sources that may not offer HD quality or be already offline while there's a more reliant source. I see it necessary to provide both sources or prefer a geo-blocked source at least in country of designation playlists.

**Comment by Dum4G** [comment]
@freearhey

**Reply by freearhey** [reply]
I am ok with the way links in playlists are sorted now, but if others think it's not the right approach, the algorithm can be changed at any time by editing the [generate.ts](https://github.com/iptv-org/iptv/blob/a4178d3c501a50e7a8fcf769b4e89b624018da32/scripts/commands/playlist/generate.ts#L107) file.

**Reply by Dum4G** [reply]
Ok, here's an example. Yesterday I have added a geoblocked source that offers some channels in FHD and leaved another source that was never geoblocked but wasn't maintained properly either. What has happened? People got only a single option that could at best be streamed in 400p but it doesn't even work. And the plot twist is that we actually have a better, working at least for some stream to bake. In a perfect world there's a dead links detection and maintainers to blame for imperfections but what if neither of them react?
